Modules/Loadable/Models/SubjectHierarchyPlugins/qSlicerSubjectHierarchyModelsPlugin.cxx

```diff
@@ -221,12 +227,142 @@ QString qSlicerSubjectHierarchyModelsPlugin::tooltip(vtkIdType itemID)const
     }
   else
     {
-    tooltipString.append(" !Invalid model!");
+    tooltipString.append(" !Invalid model");
     }
 
   return tooltipString;
 }
 
+//-----------------------------------------------------------------------------
+void qSlicerSubjectHierarchyModelsPlugin::setDisplayColor(vtkIdType itemID, QColor color, QMap<int, QVariant> terminologyMetaData)
+{
+  if (itemID == vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID)
+    {
+    qCritical() << Q_FUNC_INFO << ": Invalid input item";
+    return;
+    }
+  vtkMRMLSubjectHierarchyNode* shNode = qSlicerSubjectHierarchyPluginHandler::instance()->subjectHierarchyNode();
+  if (!shNode)
+    {
+    qCritical() << Q_FUNC_INFO << ": Failed to access subject hierarchy node";
+    return;
+    }
+
+  // Get model node and display node
+  vtkMRMLModelNode* modelNode = vtkMRMLModelNode::SafeDownCast(shNode->GetItemDataNode(itemID));
+  if (!modelNode)
+    {
+    qCritical() << Q_FUNC_INFO << ": Unable to find model node for subject hierarchy item " << shNode->GetItemName(itemID).c_str();
+    return;
+    }
+  vtkMRMLModelDisplayNode* displayNode = vtkMRMLModelDisplayNode::SafeDownCast(modelNode->GetDisplayNode());
+  if (!displayNode)
+    {
+    qCritical() << Q_FUNC_INFO << ": No display node for model";
+    return;
+    }
+
+  // Set terminology metadata
+  if (terminologyMetaData.contains(qSlicerTerminologyItemDelegate::TerminologyRole))
+    {
+    modelNode->SetAttribute(vtkSegment::GetTerminologyEntryTagName(),
+      terminologyMetaData[qSlicerTerminologyItemDelegate::TerminologyRole].toString().toLatin1().constData() );
+    }
+  if (terminologyMetaData.contains(qSlicerTerminologyItemDelegate::NameRole))
+    {
+    modelNode->SetName(
+      terminologyMetaData[qSlicerTerminologyItemDelegate::NameRole].toString().toLatin1().constData() );
+    }
+  if (terminologyMetaData.contains(qSlicerTerminologyItemDelegate::NameAutoGeneratedRole))
+    {
+    modelNode->SetAttribute("NameAutoGenerated",
```

Comment:
Without any context, "NameAutoGenerated" is a bit obscure (not clear who generates it automatically, from what, when). It would clarify things if we included SubjectHierarchy and/or Terminology in the attribute name. For example:
- SubjectHierarchy.NameAutoGenerated
- SubjectHierarchy.UseTerminologyForName
- Terminology.NodeNameAutoGenerated
- **Terminology.AutoUpdateNodeName**

I think I like the last one the most.

Reply:
Sounds good to me

Reply:
The name of the module is Terminologies (plural). Shouldn't we use that instead of singular?

Reply:
Yes, good point, we should use the exact module name.

Reply:
It would be also good to define these attribute names somewhere (in Terminologies module?) to reduce chance of typos and have a place where these attributes can be documented.